chameleonultragui/lib/gui/menu/slot_export.dart

```diff
@@ -49,14 +49,44 @@ class SlotExportMenuState extends State<SlotExportMenu> {
       );
     } else {
       CardData data = await appState.communicator!.mf1GetAntiCollData();
-      List<Uint8List> binData = [];
 
       int blockCount = mfClassicGetBlockCount(
           chameleonTagTypeGetMfClassicType(widget.slotTypes.hf));
-      for (int block = 0; block < blockCount; block += 16) {
-        Uint8List blockData =
-            await appState.communicator!.mf1GetEmulatorBlock(block, block + 16);
-        binData.add(blockData);
+
+      Uint8List binData = Uint8List(blockCount * 16);
+
+      // How many blocks to request per read command.
+      int readCount = 16;
+      int binDataIndex = 0;
+
+      for (int currentBlock = 0;
+          currentBlock < blockCount;
+          currentBlock += readCount) {
+        Uint8List result = await appState.communicator!
+            .mf1GetEmulatorBlock(currentBlock, readCount);
+
+        binData.setAll(binDataIndex, result);
+        binDataIndex += result.length;
+      }
+
+      int remainingBlocks = blockCount % readCount;
+
+      if (remainingBlocks != 0) {
+        Uint8List result = await appState.communicator!
+            .mf1GetEmulatorBlock(blockCount - remainingBlocks, remainingBlocks);
+        binData.setAll(binDataIndex, result);
+      }
+
+      List<Uint8List> sectors = [];
+      int sectorSize = 16 * 4;
+
+      for (int i = 0; i < binData.length; i += sectorSize) {
```

Comment:
I don't actually get this part. Mifare Classic block size is 16 bytes, but you are adding blocks with size 64

Reply:
One sector contains 4 blocks so 64 bytes per sector. Or should one "subarray" only contain one block?

Reply:
so the resulting array is sectors[16][64]

Reply:
resulting array must be only blocks, no sectors. Each block is 16 bytes

Reply:
ok, i will change that